Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -372,27 +372,43 @@ theorem induction_on' {P : MvPolynomial σ R → Prop} (p : MvPolynomial σ R)
     show P (monomial 0 0) from h1 0 0)
     fun _ _ _ _ha _hb hPf => h2 _ _ (h1 _ _) hPf
 
-/-- Similar to `MvPolynomial.induction_on` but only a weak form of `h_add` is required. -/
-theorem induction_on''' {M : MvPolynomial σ R → Prop} (p : MvPolynomial σ R) (h_C : ∀ a, M (C a))
+/--
+Similar to `MvPolynomial.induction_on` but only a weak form of `h_add` is required.
+In particular, this version only requires us to show
+that `M` is closed under addition of nontrivial monomials not present in the support.
+-/
+theorem monomial_add_induction_on {M : MvPolynomial σ R → Prop} (p : MvPolynomial σ R)
+    (h_C : ∀ a, M (C a))
     (h_add_weak :
```

Comment:
Can you rename these to `C` and `monomial_add` / `monomial_add_weak` / `monomial_add_of_nmem_support_of_ne_zero`, so that this works with `induction`?